### ISSUE TITLE
Inline chat for direct single edit application

### ISSUE THREAD

**Issue by marrej** [microsoft/vscode#190239]
# Utilizing the Inline chat as means for showing edits.

TLDR; The Inline chat could be cheaply extended to provide In editor edits. (PR https://github.com/microsoft/vscode/pull/190258 )

![image](https://github.com/microsoft/vscode/assets/13024177/a3ab657b-5c42-4ac9-94fa-602622819c36)

## Intro

At the moment the experimental inline chat, provides very interesting way how to display edits that are restricted to a single file (TextEdits). And assuming that this API is something that the extension authors are interested in (e.g. see https://github.com/microsoft/vscode/issues/85682 ) might provide a great addition to the extension author toolbelt.

The main benefits being (compared to Refactor Preview), that for small edits/single file edits:

- the user focus is not broken by providing unnecessarily big context
- suggested edits are editable
- Reduces the amount of clicks the user needs, to view edits

At the same time, I realise that the Inline chat purpose is to **Chat**, but at the moment it seems that the edit preview logic is fused with the internals of the chat and can not be easily extracted.

## Proposal

Extend the existing `inlineChat.start` (or introduce a new command e.g. `inlineChat.showEdit`) which would allow to take in `edits: vscode.TextEdit[]` (or vscode.WorkspaceEdit, and add additional checks in the inlineChat). The edits would be added to the options (which are passed to the `InlineChatController`), and would modify the state flow a bit.
After `INIT_UI`, break the `_nextState` and Attach the edit as a new response, then continue the `_nextState` with `APPLY_EDIT` 
If edits are provided in the options, then hide parts of the toolbar (the conversational pieces), and keep only the actions.

Done in PR https://github.com/microsoft/vscode/pull/190258

### Additional nice to have

- Allow to accept/reject the edit via command/api invocation from extension (only for the extension provided provider)
- Ability to add Markdown info text inside the Action bar (e.g. to provide some helpful hint for the edit)
- Ability to explicitly select visualisation mode (inline/side-by-side)

VS Code version: 1.81.1 
OS version: ChromeOS  114.0.5735.239

**Comment by luhc228** [comment]
Will this PR merge in the future?

**Comment by marrej** [comment]
happy to update the PR to the latest version if @jrieken will agree to have it enabled.
Although lets note that this is not the clean way how this should be done.
Rather the functionality should be exposed via a dedicated API and best have the Inline diff totally separated (as a whole new Core Component).

That being said, the dedicated API could be added easily, with the provider being registered automatically for extensionHost (While allowing only the mentioned command to be called) as intermediate step, allowing then to switch the functionality in the background of the API to a separate inline Diff with edit panel.

(Happy to add that in if required)

**Comment by jimmy-zhening-luo** [comment]
1. Nice proposal, would love to see a dedicated API for inline edits and the nice code editing interface sugar that it would enable.

1. Is this at all related to the `editor.experimentalInlineEdit.enabled` setting that I've seen recently? The reason I ask is because a bunch of conflicting `System` Keyboard Shortcuts (e.g. `Ctrl + Space`) for this experimental feature were silently added to my user settings during some update, quite annoying. I wouldn't be asking if I could find *any* public release notes on this experimental feature. Would love if someone from MS could explain what `experimentalInlineEdit` is and possibly respond to @marrej .

```jsonc
{
  // Controls whether to show inline edits in the editor.
  "editor.experimentalInlineEdit.enabled": false,
}
```

**Comment by jrieken** [comment]
> Is this at all related to the editor.experimentalInlineEdit.enabled setting that I've seen recently?

No

**Comment by sameemqiitm** [comment]
Any workarounds/examples how to implement this???

**Comment by qiyx2019** [comment]
> easily

how to use inline-chat in my vscode-ext?
